DigitalLearningSolutions.Data/Services/UserService.cs

```diff
@@ -20,6 +22,60 @@ public UserService(IDbConnection connection)
             this.connection = connection;
         }
 
+        public AdminUser? GetAdminUserByUsername(string username)
+        {
+            AdminUser? user = connection.Query<AdminUser>(
+                @"SELECT
+                        au.AdminID,
+                        au.CentreID, 
+                        ct.CentreName,
+                        au.Email,
+                        au.Forename,
+                        au.Surname,
+                        au.Password,
+                        au.CentreAdmin,
+                        au.IsCentreManager,
+                        au.ContentCreator,
+                        au.ContentManager,
+                        au.PublishToAll,
+                        au.SummaryReports,
+                        au.UserAdmin,
+                        au.CategoryID,
+                        au.Supervisor,
+                        au.Trainer,
+                        au.IsFrameworkDeveloper
+                    FROM AdminUsers AS au
+                    INNER JOIN Centres AS ct ON ct.CentreID = au.CentreID
+                    WHERE au.Active = 1 AND au.Approved = 1 AND (au.Login = @username OR au.Email = @username)",
+                new { username }
+            ).FirstOrDefault();
+
+            return user;
+        }
+
+        public List<DelegateUser> GetDelegateUsersByUsername(string username)
+        {
+            List<DelegateUser> users = connection.Query<DelegateUser>(
+                @"SELECT
+                        cd.CandidateID,
+                        cd.CandidateNumber,
+                        ct.CentreName,
+                        cd.CentreID,
+                        cd.EmailAddress,
+                        cd.FirstName,
+                        cd.LastName,
+                        cd.Password,
+                        cd.Approved
+                    FROM Candidates AS cd
+                    INNER JOIN Centres AS ct ON ct.CentreID = cd.CentreID 
+                    WHERE cd.Active = 1 AND
+                         (cd.CandidateNumber = @username OR cd.EmailAddress = @username OR cd.AliasID = @username)",
```

Comment:
Ah, is the alias used in the old system? (I might ask Kevin about where it's used across the system and how it works if so - to make sure we don't get into issues when handling this)

Reply:
It is indeed used in the old system, used identically as it is here to get delegate records by userid/email.